LEAF_Request_Portal/form.php

```diff
@@ -3061,6 +3066,24 @@ public function query($inQuery)
             }
         }
 
+        if ($joinRecordResolutionBy) {
+            require_once 'VAMC_Directory.php';
+            $dir = new VAMC_Directory;
+
+            $res2 = $this->db->prepared_query('SELECT recordID, userID as resolvedBy FROM action_history
+                                                LEFT JOIN dependencies USING (dependencyID)
+                                                LEFT JOIN actions USING (actionType)
+                                                WHERE recordID IN (' . $recordIDs . ')
+                                                and stepID > 0
+                                                ORDER BY time', array());
+
```

Comment:
As Pete stated, this current query only gets the last person to touch a request in the workflow.  That "touch" could be a comment, move between steps that doesn't complete the request, or a move backward in the workflow.  This will need to be redone so it's checking against workflow next Steps vs. last step taken.  If they are both zero, then it's at the end of the workflow and is resolved.

Reply:
The query can even be modified slightly from the one just above this one as it deals with Resolved data and probably already has the data you need for the report (or close to it).

Reply:
query fixed to pull correct info

Reply:
Issue is still present.  the query on line 3052 from this file has the exact query you need and you only need to add the action_history table to get the userID.  you might even be able to remove some of the joins depending on the data you need.  I would review that query and see how it can be modified.  it should definitely at least have the non-submitted and deleted requests removed from it's search.